core/Period/Factory.php

```diff
@@ -55,6 +56,17 @@ public static function build($period, $date, $timezone = 'UTC')
                 return new Year($date);
                 break;
         }
+
+        $customPeriodFactories = StaticContainer::get('period.custom.factories');
```

Comment:
I thought about that, but I think would be broken if a plugin accidentally removed the ability to create one of those periods, so I thought it would be better to hard code those, to make sure they are always available. (For example would it make sense to ever remove the day period?) What do you think?